nautobot/users/filters.py

```diff
@@ -40,6 +49,64 @@ class UserFilterSet(BaseFilterSet):
         to_field_name="name",
         label="Group (name)",
     )
+    has_changes = RelatedMembershipBooleanFilter(
+        field_name="changes",
+        label="Has Changes",
+    )
+    changes = NaturalKeyOrPKMultipleChoiceFilter(
+        to_field_name="user_name",
+        queryset=ObjectChange.objects.all(),
+        label="Object Changes (ID or user_name)",
+    )
+    has_logentry = RelatedMembershipBooleanFilter(
+        field_name="logentry_set",
+        label="Has Changes",
+    )
+    # TODO(timizuo): Dont know what the `to_field_name` attribute value should be yet
+    logentry = NaturalKeyOrPKMultipleChoiceFilter(
+        field_name="logentry_set",
+        # to_field_name="...",
+        queryset=LogEntry.objects.all(),
+        label="Log Entry (ID)",
+    )
+    has_object_permissions = RelatedMembershipBooleanFilter(
+        field_name="object_permissions",
+        label="Has Changes",
+    )
+    object_permissions = NaturalKeyOrPKMultipleChoiceFilter(
+        to_field_name="name",
+        queryset=ObjectPermission.objects.all(),
+        label="Object Permission (ID or name)",
+    )
+    has_rack_reservations = RelatedMembershipBooleanFilter(
+        field_name="rackreservation_set",
+        label="Has Changes",
+    )
+    # TODO(timizuo): Dont know what the `to_field_name` attribute value should be yet
```

Comment:
RackReservation doesn't have a clear primary key yet - so this should probably just be a ModelMultipleChoiceFilter for now.